### ISSUE TITLE
BUG: unified_server.cpp hardcoded stack buffers risk silent data corruption

### ISSUE THREAD

**Issue by bong-water-water-bong** [1bit-MONSTER/1bit-MONSTER#262]
## Summary

`tools/unified_server.cpp` uses hardcoded stack buffer sizes (`float hidden_buf[2048]`, `new float[262272]`) that assume a specific model architecture (H=2048, V=262272) rather than using the runtime model config. If a different model is loaded, this causes truncation, buffer overflow, or silent garbage output.

### 1. Hardcoded H=2048 stack buffer (high risk)
`tools/unified_server.cpp` lines ~396-397:
```cpp
float hidden_buf[2048];  // ZAYA_H
```
Used with:
```cpp
mgr.forward(last_token, hidden_buf)
```

If the loaded model has `hidden_size > 2048`, the `forward()` backend writes past the end of the stack buffer → **stack buffer overflow** (silent corruption or crash).

If the loaded model has `hidden_size < 2048`, the code works but wastes space — no crash, but the mismatch indicates the code wasn't designed for multi-model support.

### 2. Hardcoded V=262272 heap allocation (medium risk)
Lines ~408, 416:
```cpp
float* logits_buf = new float[262272];
```
If vocab is smaller, this works (just oversized). If vocab is larger → heap buffer overflow.

### 3. Hardcoded V=262272 in loop bounds (high risk)
Lines ~420, 423:
```cpp
for (int v = 0; v < 262272; v++)
```
If the actual model has a smaller vocab, this reads past the valid logit data but within the allocation — reads garbage, produces wrong softmax/token_logprob. If larger vocab → overflow.

## Fix

Use model config from BackendManager:
```cpp
const auto& cfg = mgr.active_info()->instance->cfg;
// or pull from BackendManager's stored config
std::vector<float> hidden_buf(cfg.hidden_size);
std::vector<float> logits_buf(cfg.vocab_size);
```

This also eliminates `new float[]` / `delete[]` churn on every token (double win: correctness + perf).
